src/test/java/IMDbTop250Scraper.java

```diff
@@ -0,0 +1,106 @@
+import java.util.HashSet;
+import java.util.Set;
+import static com.codeborne.selenide.Condition.*;
+import static com.codeborne.selenide.Selenide.*;
+
+public class IMDbTop250Scraper {
+
+    public static void main(String[] args) {
+        try {
+            openIMDbHomePage();
+            navigateToTop250();
+            scrapeTopMovies();
+        } catch (Exception e) {
+            handleError(e);
+        }
+    }
+
+    private static void scrapeTopMovies() {
```

Comment:
Still looks messy to me.. I will add my option for you and you can investigate.

Reply:
Your code working much better than my)))